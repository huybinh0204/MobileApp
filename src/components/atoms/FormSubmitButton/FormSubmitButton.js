import React from 'react';
import { useFormContext } from 'react-hook-form';
import MainButton from '_components/atoms/MainButton/MainButton';
import PropTypes from 'prop-types';

const FormSubmitButton = ({ onSubmit, ...rest }) => {
  const { handleSubmit } = useFormContext();

  return <MainButton onPress={handleSubmit(onSubmit)} {...rest} />;
};

FormSubmitButton.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default FormSubmitButton;
