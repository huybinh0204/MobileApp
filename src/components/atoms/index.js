export { default as AccountNumber } from './AccountNumber/AccountNumber';
export { default as Badge } from './Badge/Badge';
export { default as Balance } from './Balance/Balance';
export { default as BaseFormInput } from './BaseFormInput/BaseFormInput';
export { default as BaseText } from './BaseText/BaseText';
export { default as BottomSheet } from './BottomSheet/BottomSheet';
export { default as Card } from './Card/Card';
export { default as CurrencyInput } from './CurrencyInput/CurrencyInput';
export { default as Dash } from './Dash/Dash';
export { default as Form } from './Form/Form';
export { default as FormInput } from './FormInput/FormInput';
export { default as FormSubmitButton } from './FormSubmitButton/FormSubmitButton';
export { default as GeneralContainer } from './GeneralContainer/GeneralContainer';
export { default as IconSvg } from './IconSvg/IconSvg';
export { default as Link } from './Link/Link';
export { default as MainButton } from './MainButton/MainButton';
export { default as Money } from './Money/Money';
export { default as PdfView } from './PdfView/PdfView';
export { default as Plaid } from './Plaid/Plaid';
export { default as PressableListItem } from './PressableListItem/PressableListItem';
export { default as ProgressBar } from './ProgressBar/ProgressBar';
export { default as SegmentedTextInput } from './SegmentedTextInput/SegmentedTextInput';
export { default as SelectButton } from './SelectButton/SelectButton';
export { default as Switch } from './Switch/Switch';
export { default as TabItem } from './TabItem/TabItem';
export { default as Toast } from './Toast/Toast';
export { default as TransactionItem } from './TransactionItem/TransactionItem';
