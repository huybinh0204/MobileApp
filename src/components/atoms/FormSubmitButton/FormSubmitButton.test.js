import React from 'react';
import { render } from '@testing-library/react-native';
import { useForm } from 'react-hook-form';
import { Form } from '_components/atoms';
import Theme from '_utilities/Theme';
import FormSubmitButton from './FormSubmitButton';

describe('FormSubmitButton', () => {
  let props;

  const RenderFormSubmit = (innerProps) => {
    const formMethods = useForm({});

    return (
      <Theme>
        <Form formMethods={formMethods}>
          <FormSubmitButton {...innerProps} />
        </Form>
      </Theme>
    );
  };

  beforeEach(() => {
    props = {
      children: 'test',
      onSubmit: jest.fn(),
    };
  });

  describe('structure', () => {
    it('should match the snapshot', () => {
      const { toJSON } = render(<RenderFormSubmit {...props} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
