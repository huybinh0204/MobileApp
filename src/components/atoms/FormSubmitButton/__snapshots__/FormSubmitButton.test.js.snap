// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSubmitButton structure should match the snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#3D3397",
      "borderRadius": 10,
      "opacity": 1,
      "paddingHorizontal": 45.5,
      "paddingVertical": 31,
    }
  }
>
  <Text
    allowFontScaling={false}
    disabled={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Poppins-SemiBold",
          "fontSize": 25.5,
          "lineHeight": 35.5,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
